plaso/parsers/ios_lockdownd.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+"""Parser for the logs of the iOS lockdown daemon."""
+
+import pyparsing
+
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import errors
+from plaso.lib import definitions
+from plaso.parsers import manager
+from plaso.parsers import text_parser
+
+
+class IOSLockdownLogData(events.EventData):
+  """iOS lockdownd event data.
+
+  Attributes:
+    body (str): body of the event line.
+    process_identifier (str): process making the request to lockdownd.
+  """
+
+  DATA_TYPE = 'ios:lockdown:log:line'
+
+  def __init__(self):
+    """Initializes iOS lockdown daemon log event data."""
+    super(IOSLockdownLogData, self).__init__(data_type=self.DATA_TYPE)
+    self.body = None
+    self.process_identifier = None
+
+
+class IOSLockdownParser(text_parser.PyparsingMultiLineTextParser):
+  """Parser for iOS lockdown daemon log files (ios_lockdownd.log"""
+
+  NAME = 'ios:lockdownd:log'
+  DATA_FORMAT = 'iOS lockdown daemon log'
+
+  TWO_DIGITS = text_parser.PyparsingConstants.TWO_DIGITS
+
+  DATE_ELEMENTS = (
+      TWO_DIGITS.setResultsName('month') + pyparsing.Suppress('/') +
+      TWO_DIGITS.setResultsName('day_of_month') + pyparsing.Suppress('/') +
+      TWO_DIGITS.setResultsName('two_digit_year'))
+
+  TIME_MSEC_ELEMENTS = text_parser.PyparsingConstants.TIME_MSEC_ELEMENTS
+
+  _TIMESTAMP = DATE_ELEMENTS + TIME_MSEC_ELEMENTS
+
+  _PID = (
+      pyparsing.Suppress('pid=') +
+      pyparsing.Word(pyparsing.nums).setResultsName('process_identifier'))
+
+  _BODY_END = pyparsing.StringEnd() | _TIMESTAMP
+
+  _BODY = pyparsing.SkipTo(_BODY_END).setResultsName('body')
+
+  _LINE_GRAMMAR = _TIMESTAMP + _PID + _BODY + pyparsing.ZeroOrMore(
+      pyparsing.lineEnd())
+
+  LINE_STRUCTURES = [('log_entry', _LINE_GRAMMAR)]
+
+  def ParseRecord(self, parser_mediator, key, structure: object):
```

Comment:
pytype information not yet supported by this code base due to version differences in pytype definitions